WordPress/Classes/Utility/Spotlight/SearchManager.swift

```diff
@@ -252,37 +252,45 @@ fileprivate extension SearchManager {
                    blogID: NSNumber,
                    onSuccess: @escaping (_ post: AbstractPost) -> Void,
                    onFailure: @escaping () -> Void) {
-        let context = ContextManager.sharedInstance().mainContext
+        let coreDataStack = ContextManager.shared
 
-        guard let blog = Blog.lookup(withID: blogID, in: context) else {
+        guard let blog = Blog.lookup(withID: blogID, in: coreDataStack.mainContext) else {
                 onFailure()
                 return
         }
 
-        let postService = PostService(managedObjectContext: context)
-        postService.getPostWithID(postID, for: blog, success: { apost in
-            onSuccess(apost)
-        }, failure: { error in
-            onFailure()
-        })
+        let postRepository = PostRepository(coreDataStack: coreDataStack)
+        Task { @MainActor in
+            do {
+                let postObjectID = try await postRepository.getPost(withID: postID, from: .ofSaved(blog))
```

Comment:
I like the `ofSaved` and `ofUnsaved` names in the implementation, but I wonder if they deliver enough clarity when used in a situation like this?

I feel 

```suggestion
                let postObjectID = try await postRepository.getPost(withID: postID, from: .identifier(forSaved: blog))
```

---

Aside: I considered using `identifierForSaved` but checked the [API guide lines ](https://www.swift.org/documentation/api-design-guidelines/#general-conventions) and they recommend against it

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/6e2c2a97-1c4b-4e40-aa1e-694df32bb691)

Reply:
I'm not a big fan of the current names, but "identifier" is duplicated in the full signature `CoreDataObjectIdentifier.identifier(forSaved:)` though...